File: content/problems/1572-matrix-diagonal-sum/solutions/two-diagonal-scan/Solution.kt
class Solution { fun diagonalSum(mat:Array<IntArray>):Int{val n=mat.size;var answer=0;for(row in 0 until n)answer+=mat[row][row]+mat[row][n-1-row];if(n%2==1)answer-=mat[n/2][n/2];return answer} }

File: content/problems/1572-matrix-diagonal-sum/solutions/two-diagonal-scan/solution.cpp
class Solution {
    public: int diagonalSum(vector<vector<int>>& mat){
        int n=mat.size(),answer=0;
        for(int row=0;row<n;++row)answer+=mat[row][row]+mat[row][n-1-row];
        if(n%2)answer-=mat[n/2][n/2];
        return answer;
    }
};
